flexget/plugins/search/1337x.py

```diff
@@ -0,0 +1,165 @@
+from __future__ import unicode_literals, division, absolute_import
+from future.moves.urllib.parse import quote
+from builtins import *  # pylint: disable=unused-import, redefined-builtin
+
+import logging
+import re
+
+from flexget import plugin, db_schema
+from flexget.entry import Entry
+from flexget.event import event
+from flexget.utils.requests import TimedLimiter, RequestException
+from flexget.utils.requests import Session as RequestSession
+from flexget.utils.soup import get_soup
+from flexget.utils.search import torrent_availability
+
+log = logging.getLogger('1337x')
+Base = db_schema.versioned_base('1337x', 0)
+
+requests = RequestSession()
+requests.add_domain_limiter(TimedLimiter('1337x.to', '5 seconds'))  # TODO find out if they want a delay
+
+class _1337x(object):
+    """
+        1337x search plugin.
+    """
+
+    schema = {
+        'oneOf': [
+            {'type': 'boolean'},
+            {
+                'type': 'object',
+                'properties': {
+                    'order_by': {'type': 'string', 'enum': ['seeders', 'leechers', 'time', 'size'], 'default': 'seeders'}
+                },
+                'additionalProperties': False
+            }
+        ]
+    }
+
+    base_url = 'http://1337x.to/'
+    errors = False
+
+    def get(self, url, params):
+        """
+        Wrapper to allow refreshing the cookie if it is invalid for some reason
+        :param url:
+        :param params:
+        :return:
+        """
+        cookies = None
+
+        response = requests.get(url, params=params, cookies=cookies)
+
+        return response
+
+    # urlrewriter API
+    def url_rewritable(self, task, entry):
+        url = entry['url']
+        if url.endswith('.torrent'):
+            return False
+        if url.startswith('http://1337x.to/'):
+            return True
+        return False
+
+    def url_rewrite(self, task, entry):
+        """
+            Gets the download information for 1337x result
+        """
+
+        url = entry['url']
+
+        log.info('1337x rewriting download url: %s' % url)
+
+        try:
+            page = self.get(url, None)
+            log.debug('requesting: %s', page.url)
+        except RequestException as e:
+            log.error('1337x request failed: %s', e)
+
+        soup = get_soup(page.content)
+
+        # Get the infohash - not needed but the code is here *shrugs*
+        #infohash = str(soup.find("div", class_="infohash-box").contents)
+        #infohash = infohash[-(len(infohash)-infohash.find(':'))+2:]
+        #infohash = infohash[:infohash.find(' ')]
+
+        magnetURL = str(soup.find("a", id="magnetdl").get('href')).lower()
+        torrentURL = str(soup.find("a", id="torrentdl").get('href')).lower()
+
+        # Can't update the title to the full title name because it breaks entry.remove
+        #title = str(soup.title.string).replace("Download Torrent ","").replace("| 1337x","")
+
+        entry['url'] = torrentURL
+        entry['urls'] = []
```

Comment:
Dunno if I agree with this. Maybe the new urls should be prepended instead of clearing the list?